scripts/extract-hvd-content.ts

```diff
@@ -3,32 +3,30 @@ import path from 'path'
 import { fetchGithubArchiveZip } from 'lib/fetch-github-archive-zip'
 import { isDeployPreview } from 'lib/env-checks'
 
+export const BASE_REPO_CONFIG = {
+	owner: 'hashicorp',
+	ref: process.env.CURRENT_GIT_BRANCH || 'main',
+	repo: 'hvd-docs',
+	contentPath: '/content',
+}
+
 /**
  * Set the location in the local filesystem where we'll extract HVD content.
  */
-export const HVD_CONTENT_DIR = path.join(
+const HVD_REPO_DIR = path.join(
 	process.cwd(),
 	'src/.extracted/hashicorp-validated-designs'
 )
 
-const BASE_REPO_CONFIG = {
-	owner: 'hashicorp',
-	ref: 'main',
-	repo: 'hvd-docs',
-}
-
 /**
- * For now, we extract content from `hvd-docs` only.
- *
- * TODO: provide a solution for local development of HVD content.
+ * Detect whether we are in the hashicorp/hvd-docs repo directly
+ * and if so use the local path defined in the build or start script.
  *
- * One option might be, if we detect we're in local development using
- * the `hashicorp/dev-portal` repo directly, we could attempt to load in
- * `hvd-docs` content from the local filesystem in an adjacent directory
- * one level up from the current working directory. In other words, we'd assume:
- * - `dev-portal` is the current working directory (i.e. `process.cwd()`
- * - `dev-portal/../hvd-docs` has local content being developed
+ * Otherwise, use the path generated from the `extractHvdContent` script
  */
+export const HVD_CONTENT_DIR =
+	process.env.LOCAL_CONTENT_DIR ||
```

Comment:
This is where we are defining the local content path for use in the build preview.